src/com/swdc/codetime/managers/EclipseProject.java

```diff
@@ -0,0 +1,85 @@
+package com.swdc.codetime.managers;
+
+import java.io.File;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.IPath;
+
+import swdc.java.ops.manager.UtilManager;
+import swdc.java.ops.model.Project;
+import swdc.java.ops.providers.IdeProject;
+
+public class EclipseProject implements IdeProject {
+	
+	public static String lastOpenFile = "";
+
+	@Override
+	public Project buildKeystrokeProject(Object iProj) {
+		return createProject((IProject) iProj);
+	}
+
+	@Override
+	public String getFileSyntax(File file) {
+		if (file == null || !file.exists()) {
+			return "";
+		}
+		return com.google.common.io.Files.getFileExtension(file.getAbsolutePath());
+	}
+
+	@Override
+	public Project getFirstActiveProject() {
```

Comment:
This is to handle the case when there are multiple projects open in the same workspace at the same time? Might call this method something like `getCurrentProject`.

Reply:
yeah, we can do that but its in the java ops interface with that name now. its a tedious change